src/components/MARCTable/MARCTableRow.js

```diff
@@ -123,22 +158,31 @@ export const MARCTableRow = ({
   const validateSubfield = useCallback(
     value => {
       if (actionValue === MAPPING_DETAILS_ACTIONS.ADD) {
-        return validateAlphanumericValue(value) || validateSubfieldField(fieldValue)(value);
+        return validateAlphanumericOrAllowedValue(value) || validateSubfieldField(fieldValue)(value);
+      }
+
+      if (actionValue === MAPPING_DETAILS_ACTIONS.DELETE) {
+        return validateRequiredField(value) || validateAlphanumericOrAllowedValue(value, '*') || validateSubfieldField(fieldValue)(value);
       }
 
       return null;
     },
     [actionValue, fieldValue],
   );
 
-  const handleActionChange = e => {
-    const { ADD } = MAPPING_DETAILS_ACTIONS;
+  const fillFieldsWithDefaultValue = (items, valueToFill) => items
+    .map(item => !item.value && dispatchOnChange(MAPPING_DETAILS_FORM_NAME, item.path, valueToFill));
```

Comment:
Could you please clarify what is the return value from `dispatchOnChange` is going to be?

Reply:
Refactored. Main purpose - on Delete action fill empty indicator1, indicator2 and subfield fields in appropriate row with asterisk